File: python/examples/serve.py
"""Serve the Python example calendar (the same service as ../../examples/calendar.ts).

    PARLEY_TRUST=ed25519:... uv run python examples/serve.py
    calendar → parley://127.0.0.1:7457  and  http://127.0.0.1:8457/parley

PARLEY_TRUST is a comma-separated list of trusted principal keys. --stdio serves NDJSON on
stdin/stdout instead (for ``stdio:`` URLs).
"""

from __future__ import annotations

import asyncio
import os
import sys

from calendar_example import calendar

from parley import serve_http, serve_stdio, serve_tcp


async def main() -> None:
    trust = [k for k in os.environ.get("PARLEY_TRUST", "").split(",") if k]
    host = os.environ.get("HOST", "127.0.0.1")
    tcp_port = int(os.environ.get("PARLEY_PORT", "7457"))
    http_port = int(os.environ.get("PARLEY_HTTP_PORT", "8457"))
    cal = calendar(trust)
    if "--stdio" in sys.argv:
        await serve_stdio(cal)
        return
    servers = [await serve_tcp(cal, host, tcp_port), await serve_http(cal, host, http_port)]
    print(
        f"parley python example up · calendar parley://{host}:{tcp_port} http://{host}:{http_port}/parley"
        f" · trusting {len(trust)} principal(s)",
        file=sys.stderr, flush=True,
    )
    await asyncio.gather(*(s.serve_forever() for s in servers))


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        pass

File: examples/serve.ts
// Serve the example services.
//   calendar → yea://127.0.0.1:7447   and  http://127.0.0.1:8447/yea
//   shop     → yea://127.0.0.1:7449   and  http://127.0.0.1:8449/yea
// Trusted principals come from YEA_TRUST (comma-separated "ed25519:…" keys).
import { listen, serveHttp } from '@yea-protocol/sdk/node';
import { calendar } from './calendar.ts';
import { shop } from './shop.ts';

const trust = (process.env.YEA_TRUST ?? '').split(',').filter(Boolean);
const host = process.env.HOST ?? '127.0.0.1';
const cal = calendar({ trust });
const sh = shop({ trust });

await Promise.all([
  listen(cal, { port: 7447, host }),
  serveHttp(cal, { port: 8447, host }),
  listen(sh, { port: 7449, host }),
  serveHttp(sh, { port: 8449, host }),
]);
console.error(
  `yea examples up · calendar yea://${host}:7447 http://${host}:8447/yea · shop yea://${host}:7449 http://${host}:8449/yea · trusting ${trust.length} principal(s)`,
);
